### ISSUE TITLE
Add a way to create roles and databases in each PG cluster using Ansible inventory

### ISSUE THREAD

**Issue by drybjed** [debops/ansible-postgresql#1]
Roles and databases in PostgreSQL clusters can be created using `postgresql_user` and `postgresql_db` Ansible modules, which is great for application roles, but not very useful if you want to have custom roles or databases on your servers (unless you write a specific playbook for that).

It would be useful if possible, to add some tasks that can create roles and databases defined in each cluster definition in inventory.

**Comment by scosist** [comment]
I started taking a stab at this but then I got lost on the expected structure of custom roles and databases for postgresql. Initially, I was modelling these after your implementation for mysql with `mysql_databases` and `mysql_users` but then I noticed where you mention here it would be useful to add tasks to create these <i>as defined</i> in each cluster definition. If I understand correctly, you are referring to `postgresql_default_cluster` and `postgresql_clusters` definitions (https://github.com/debops/ansible-postgresql/blob/master/defaults/main.yml#L105). Should I then expect to create custom roles and databases based on the following structure?
```yaml
postgresql_default_cluster:
  - name: 'main'
    port: '5432'
    hba:
      - user: 'custom_user_1'
        database: 'custom_database_1'
      - user: 'custom_user_2'
        database: 'custom_database_2'
```
In the case that custom roles and databases need to be defined by another Ansible role, rather than redefine `postgresql_default_cluster`, perhaps another name like `postgresql_default_cluster_custom` would be used. I question whether this structure is the best way to go for defining custom roles and databases. Have I understood your meaning or am I completely off base?

**Comment by drybjed** [comment]
Hi @scosist :-) After giving it some thought, I think that `debops.postgresql` role should follow in the footsteps in `debops.mariadb` and `debops.mariadb_server` roles - current PostgreSQL role would be renamed to `debops.postgresql_server`, and a new role that only focuses on role/database management on local or remote PG servers would be created. That new `debops.postgresql` role could be then uses as a role dependency without dragging the whole firewall, APT preferences and other things with it.

One issue I had to resolve was the management of `pg_hba` and `pg_indent` files in the cluster. I wanted to make them manageable from the "client" role, but it looks like it's a very complex thing to do:

- you have multiple data sources to consider - client role and server role with the defaults;

- `pg_hba` entries need to be written with a specific order and both roles would need to manage that order correctly in unison to make this idempotent;

- when you have 1 central PG server and multiple clients managing it, having one `pg_hba` file edited by multiple hosts at once would either break things or make multiple entries, depending on the method used;

- going the `assemble` route to fix the above problem would also create multiple redundant entries very easily;

So, taking all of that into account, I think that the best course of action at the moment would be to leave the `pg_hba` and `pg_indent` configuration at the server side. An admin would be expected to correctly configure the entries before setting up remote access to the server, so that client role could create roles and databases which then could be used immediately afterwards.

To summarize - I plan to clean current `debops.postgresql` role up and rename it to `debops.postgresql_server`, and create a completely new PG client role from scratch to manage the roles and databases.

**Comment by drybjed** [comment]
The PostgreSQL roles can now create users and databases via inventory, therefore I'm closing this issue as resolved.
